src/mahoji/lib/abstracted_commands/blackjackCommand.ts

```diff
@@ -0,0 +1,211 @@
+
+const returnButtons = [
+	new ActionRowBuilder<ButtonBuilder>().addComponents([
+		new ButtonBuilder({
+			label: 'Hit',
+			style: ButtonStyle.Secondary,
+			customId: 'hit'
+		}),
+		new ButtonBuilder({
+			label: 'Stand',
+			style: ButtonStyle.Secondary,
+			customId: 'stand'
+		})
+	])
+];
+
+interface Card {
+	suit: string;
+	value: string;
+}
+
+// Define constants for suits and values
+const suits = ['Hearts', 'Diamonds', 'Clubs', 'Spades'];
+const values = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'Jack', 'Queen', 'King', 'Ace'];
+
+// Function to create a deck of cards
+function createDeck(): Card[] {
+	const deck: Card[] = [];
+	for (const suit of suits) {
+		for (const value of values) {
+			deck.push({ suit, value });
+		}
+	}
+	return deck;
+}
+
+// Function to shuffle the deck
+function shuffleDeck(deck: Card[]): void {
+	for (let i = deck.length - 1; i > 0; i--) {
+		const j = Math.floor(Math.random() * (i + 1));
+		[deck[i], deck[j]] = [deck[j], deck[i]];
+	}
+}
+
+// Function to deal a card from the deck
+function dealCard(deck: Card[]): Card {
+	return deck.pop()!;
+}
+
+// Function to calculate the value of a hand
+function calculateHandValue(hand: Card[]): number {
+	let sum = 0;
+	let aceCount = 0;
+	for (const card of hand) {
+		if (card.value === 'Ace') {
+			sum += 11;
+			aceCount++;
+		} else if (card.value === 'Jack' || card.value === 'Queen' || card.value === 'King') {
+			sum += 10;
+		} else {
+			sum += parseInt(card.value);
+		}
+	}
+	while (sum > 21 && aceCount > 0) {
+		sum -= 10;
+		aceCount--;
+	}
+	return sum;
+}
+
+export async function blackjackCommand(
+	user: MUser,
+	_amount: string,
+	interaction: ChatInputCommandInteraction
+): CommandResponse {
+	await deferInteraction(interaction);
+	const amount = mahojiParseNumber({ input: _amount, min: 1 });
+
+	if (user.isIronman) {
+		return "Ironmen can't gamble! Go pickpocket some men for GP.";
+	}
+
+	if (!amount) {
+		return `**Blackjack:**
+- Beat the dealer to win! You must gamble between 20m and 1b.`;
+	}
+
+	if (amount < 20_000_000 || amount > 1_000_000_000) {
+		return 'You can only gamble between 20m and 1b.';
+	}
+
+	const channel = globalClient.channels.cache.get(interaction.channelId);
+	if (!channelIsSendable(channel)) return 'Invalid channel.';
+
+	await handleMahojiConfirmation(
+		interaction,
+		`Are you sure you want to gamble ${toKMB(amount)}? You might lose it all, you might double it.`
+	);
+	await user.sync();
+	const currentBalance = user.GP;
+	if (currentBalance < amount) {
+		return "You don't have enough GP to make this bet.";
+	}
+	await user.removeItemsFromBank(new Bank().add('Coins', amount));
+
+	const winnings = amount * 2;
+
+	// Create and shuffle the deck
+	const deck = createDeck();
+	shuffleDeck(deck);
+
+	// Deal initial hands
+	const playerHand = [dealCard(deck), dealCard(deck)];
+	const dealerHand = [dealCard(deck), dealCard(deck)];
+
+	// Check for player blackjack
+	if (calculateHandValue(playerHand) === 21) {
```

Comment:
Blackjack should pay 2.5x  (1.5x their bet in profit). 

Or a push if dealer also has blackjack. 

You could get crazy with insurance, even-money on blackjack vs A, etc, but that's probably not necessary :D